.github/workflows/build_test.yml

```diff
@@ -0,0 +1,55 @@
+name: Build/Test for PR and collaborator push
+
+on:
+  # allows us to run workflows manually
+  workflow_dispatch:
+  pull_request:
+    paths-ignore:
+      - '.github/workflows/build_test_publish.yml'
+      - 'docker/**'
+      - 'doc/**'
+      - 'CHANGELOG.rst'
+  push:
+    paths-ignore:
+      - '.github/workflows/build_test_publish.yml'
+      - 'docker/**'
+      - 'doc/**'
+      - 'CHANGELOG.rst'
+
+jobs:
+  build-and-test:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ubuntu_versions : [
+          20.04,
+          22.04,
+        ]
+        pkg_mgr : [
+          apt,
+          conda,
+        ]
+        cycamore_tag: [
+          latest,
+        ]
+    
+    container:
+      image: ghcr.io/cyclus/cycamore_${{ matrix.ubuntu_versions }}_${{ matrix.pkg_mgr }}/cycamore:${{matrix.cycamore_tag}}
+
+    steps:
+      - name: Checkout Cymetric
+        uses: actions/checkout@v3
+
+      - name: Build Cymetric
+        run: |
+          apt update && apt install -y python3-pip
```

Comment:
For now I added another step in the workflow.  Eventually in cyclus we should use the `python -m pip install` method to install cyclus instead of just calling `setup.py` directly, so we will need pip as a dependency anyway

Reply:
I've implemented this in a branch of cyclus that I have, along with some updates to the cmake build.  I stumbled along some fixes while trying to get cython working, but I can make a separate PR for these general build changes